go/libkb/merkle_client.go

```diff
@@ -63,10 +65,23 @@ func (h1 NodeHashShort) bytes() []byte {
 	return h1[:]
 }
 
+func (h1 NodeHashShort) IsNil() bool {
+	return false
+}
+
+func (h1 NodeHashShort) Eq(h2 NodeHash) bool {
+	return bytes.Equal(h1.bytes(), h2.bytes())
+}
+
 func (h1 NodeHashShort) ExportToHashMeta() keybase1.HashMeta {
 	return keybase1.HashMeta(h1.bytes())
 }
 
+func (h1 NodeHashLong) Check(s string) bool {
+	h2 := sha512.Sum512([]byte(s))
+	return FastByteArrayEq(h1[:], h2[:])
```

Comment:
also i'd use a constant-time comparison

Reply:
I see.  While you're here, I'd change it to constant time. Seems dangerous to short-circuit when comparing hashes.